pkg/broker/privateendpoint/privateendpoint.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package privateendpoint
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"path"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2020-11-01/network"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/pkg/errors"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+type (
+	Provider = string
+	Region   = string
+)
+
+type PrivateEndpoints map[Provider]map[Region]*EndpointService
+
+type EndpointService struct {
+	ID        string             `json:"serviceID,omitempty"`
+	Endpoints []*PrivateEndpoint `json:"endpoints,omitempty"`
+}
+
+type PrivateEndpoint struct {
+	SubscriptionID     string `json:"subscriptionID,omitempty"`
+	ResourceGroup      string `json:"resourceGroup,omitempty"`
+	VirtualNetworkName string `json:"virtualNetworkName,omitempty"`
+	SubnetName         string `json:"subnetName,omitempty"`
+	EndpointName       string `json:"endpointName,omitempty"`
+}
+
+// TODO: this is a hack to get Azure working ASAP
+// replace with proper auth (see below)
+type tokenProvider struct{}
+
+func (*tokenProvider) OAuthToken() string {
+	return os.Getenv("AZURE_BEARER_TOKEN")
```

Comment:
To get the token run `az account get-access-token`
The token expires in 1 hour (barely enough for testing).